src/databricks/labs/dqx/check_funcs.py

```diff
@@ -1616,11 +1737,20 @@ def _validate_ref_params(
         raise ValueError("The number of columns to check against the reference columns must be equal.")
 
 
+def _does_not_match_pattern(column: Column, pattern: DQPattern) -> Column:
+    """
+    Internal function that returns a Boolean Column indicating if values
+    in the column do NOT match the given pattern.
+    """
+    _, _, col_expr = _get_normalized_column_and_expr(column)
```

Comment:
Better to create a separate function for getting the col_expr, instead of ignoring fields. I think it could also be used in some other checks.